### PR TITLE
better error message for immutable edits to builds

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
new error is now:
# Build ruby-sample-build-1 was not valid:
# * invalid value 'DockerImage2', Details: the source of a builder image must be an 'ImageStreamTag', 'ImageStreamImage', or 'DockerImage'
# * invalid value 'build ruby-sample-build-1', Details: spec is immutable
#

**Comment by bparees** [comment]
@liggitt ptal as the person most interested in field errors :)

this is for https://bugzilla.redhat.com/show_bug.cgi?id=1240580

File: pkg/build/api/validation/validation.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) fielderro
 	allErrs = append(allErrs, ValidateBuild(build)...)
 
 	if !kapi.Semantic.DeepEqual(build.Spec, older.Spec) {
-		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec", build.Spec, "spec is immutable"))
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec", "build "+build.ObjectMeta.Name, "spec is immutable"))
```
Comment (mfojtik) [inline pkg/build/api/validation/validation.go]: I think you can do just `build.Name`

Comment (rhcarvalho) [inline pkg/build/api/validation/validation.go]: yup +1

Comment (liggitt) [inline pkg/build/api/validation/validation.go]: for reference, this is what an attempt to update an immutable pod spec does:
```
allErrs = append(allErrs, errs.NewFieldInvalid("spec", "content of spec is not printed out, please refer to the \"details\"", "may not update fields other than container.image"))
```

Comment (bparees) [inline pkg/build/api/validation/validation.go]: @liggitt bothers me that that message doesn't tell you which pod the spec is from...  is it safe to assume there will always be enough surrounding context to tell the user which pod they were trying to modify?  (and that it was a pod, for that matter)?

if so i'm fine w/ that message.

Comment (liggitt) [inline pkg/build/api/validation/validation.go]: for normal fields, there's minimal context as well... I would hope whatever rolls up the errors tells you the overarching object that failed... if not, that's probably the right place to put identifying info, not in the guts of a field error

**Comment by bparees** [comment]
@liggitt message updated.

**Comment by liggitt** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Pull request merged**